### PR TITLE
Adding Delete packages feature

### PR DESCRIPTION
**Pull request by ann-mwangi**
This is a pull request to add a delete-package feature.

### CODE REVIEW

File: source/Octopus.Cli/Commands/Package/DeletePackage.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using Octopus.Client.Model;
+using Octopus.CommandLine;
+using Octopus.CommandLine.Commands;
+using Octopus.Versioning.Octopus;
+namespace Octopus.Cli.Commands.Releases
+{
+    [Command("delete-package", Description = "Deletes a package from the built-in NuGet repository in an Octopus Server.")]
+    public class DeletePackageCommand : ApiCommand, ISupportFormattedOutput
+    {
+        static readonly OctopusVersionParser OctopusVersionParser = new OctopusVersionParser();
```
Comment (rain-on) [inline source/Octopus.Cli/Commands/Package/DeletePackage.cs:18]: nit: doesn't appear to be used, can it be removed?

File: source/Octo.Tests/Commands/DeletePackageCommandFixture.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                CommandOutputProvider);
+
+            packageId = Guid.NewGuid().ToString();
+            
+        }
+
+        [Test]
+        public async Task DefaultOutput_ShouldDeleteTheGivenPackage()
+        {
+            deletePackageCommand.PackageName = packageId;
```
Comment (rain-on) [inline source/Octo.Tests/Commands/DeletePackageCommandFixture.cs:33]: is it a name, or an id?

File: source/Octo.Tests/Commands/DeletePackageCommandFixture.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            Repository.BuiltInPackageRepository.GetPackage(packageId, null)
+            .Returns(new PackageFromBuiltInFeedResource{Id = packageId});
+            
+            await deletePackageCommand.Execute(CommandLineArgs.ToArray()).ConfigureAwait(false);
+
+            LogLines.Should().Contain($"Deleting package");
```
Comment (rain-on) [inline source/Octo.Tests/Commands/DeletePackageCommandFixture.cs:39]: Can we test that we tried to send a HTTP request, or that the package no longer exists?

File: source/Octopus.Cli/Commands/Package/DeletePackage.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            var options = Options.For("Deletion");
+            options.Add<string>("package=", "Name of the package.", v => PackageName = v);
+        }
+        public string PackageName { get; set; }
+        public async Task Request()
+        {
+            if (string.IsNullOrWhiteSpace(PackageName)) throw new CommandException("Please specify a package name using the parameter: --package=XYZ");
+
+            var package = await Repository.BuiltInPackageRepository.GetPackage(PackageName, null).ConfigureAwait(false);
+            await Repository.BuiltInPackageRepository.DeletePackage(package).ConfigureAwait(false);
```
Comment (rain-on) [inline source/Octopus.Cli/Commands/Package/DeletePackage.cs:31]: does this still behave if the package doesn't exist? (and thus I can't delete it?)

File: source/Octo.Tests/Commands/DeletePackageCommandFixture.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        }
```
Comment (rain-on) [inline source/Octo.Tests/Commands/DeletePackageCommandFixture.cs:40]: Is it worth adding a test to prove the exception is thrown if the package-name is null and/or empty?

**Comment by ann-mwangi** [comment]
🤦 should have ensured that I have pushed all my changes before.

File: source/Octopus.Cli/Commands/Package/DeletePackage.cs (hunk 1 above)
Comment (ann-mwangi) [reply source/Octopus.Cli/Commands/Package/DeletePackage.cs]: Done

File: source/Octo.Tests/Commands/DeletePackageCommandFixture.cs (hunk 5 above)
Comment (ann-mwangi) [reply source/Octo.Tests/Commands/DeletePackageCommandFixture.cs:41]: I have added some tests.

**Review by rain-on — APPROVED** [review]
LGTM!
